rp2040-hal/src/i2c.rs

```diff
@@ -230,29 +230,47 @@ valid_pins! {
     }
 }
 
-/// Operational mode of the I2C peripheral.
+/// Operational mode of the I2C Block.
```

Comment:
Do you think this is an improvement? I'd prefer "peripheral". (Very subjective, of course!)

Reply:
I renamed these to remove ambiguity with peripheral the IP block in the rp2040 chip and peripheral the operational mode of the IP block.

I don't have a strong opinion about IP Block/Peripheral/Controller in place of Peripheral/Slave/Master, feel free to offer alternatives :)

Reply:
Good point. I'll think about it.

Reply:
What about Controller / Target instead of Master / Slave? Then Peripheral doesn't need renaming.
Controller / Target seems to be the terminology chosen by NXP (successor of Philips, inventor of I2C) so it's as official as it can get.
https://community.nxp.com/t5/Other-NXP-Products/Alternative-inclusive-terms-for-master-and-slave-in-I2S-I2C/m-p/1245101#M10574
The same words are used by wikipedia, https://en.wikipedia.org/wiki/I%C2%B2C

Reply:
Sounds good to me.
That covers the operation modes. Should we rename `Peripheral` into `Target`?

What about the IP Block ? Is this what we should call `peripheral`?

Reply:
Exactly. I think that way we can minimize confusion.

Reply:
created https://github.com/rp-rs/rp-hal/issues/752 to track the change.